Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -225,9 +230,43 @@ public void run(@NotNull ProgressIndicator indicator) {
     @Override
     public void beforeDocumentChange(DocumentEvent e) {
       myServerData.onDocumentChanged(e);
+      synchronized (myLock) {
+        final Document document = e.getDocument();
+        myUpdatedDocuments.add(document);
+        scheduleUpdateFilesContent();
+      }
     }
   };
 
+  private void scheduleUpdateFilesContent() {
```

Comment:
DartServerData contains data for all editors, visible and not visible. It clears file data on editor tab close.
Subscriptions are made for visible files only.
If we get doc change for not visible editor - yes, we will not get new data, neither before onDocumentUpdate(), nor after, so cached ranges will be shifted and won't break (but may be incorrect until file opened and new data arrives).

Looks like the problem that we are trying to solve is actual for visible files only. Will `if(visible)` be a correct `if` in beforeDocumentChange()?